### PR TITLE
updated default name

### PR DESCRIPTION
**Pull request by grace-harper-ibm**
### What are the issues this pull addresses (issue numbers / links)?
#579 

### Did you add tests to cover your changes (yes/no)?
No

### Did you update the documentation accordingly (yes/no)?
No

### Did you read the CONTRIBUTING document (yes/no)?
Yes

### Summary
Default param name for "name" for QComponent is class name

### CODE REVIEW

File: qiskit_metal/_gui/widgets/create_component_window/parameter_entry_window.py [hunk 1]
```diff
@@ -442,8 +443,10 @@ class definition
                 return memtup[1]
 
 
-def create_default_from_type(my_t: type):
+def create_default_from_type(my_t: type, param_name=None):
     """Create default values for a given type"""
+    if param_name is not None:
+        return param_name + "-" + str(random.randint(0, 1000))
```
Comment (JeremyDrysdale-ibm) [inline qiskit_metal/_gui/widgets/create_component_window/parameter_entry_window.py:449]: Below these lines in the list of `if` statements, I still see two instances of `fake-param` being returned.

Comment (grace-harper-ibm) [reply qiskit_metal/_gui/widgets/create_component_window/parameter_entry_window.py:449]: Hi Jeremy, 
Those should stay as fake-param. 
Zlatko just wanted the "name" argument default changed.

File: qiskit_metal/_gui/widgets/create_component_window/parameter_entry_window.py [hunk 2]
```diff
@@ -255,7 +255,7 @@ def _setup_source(self):
 
     @QComponentParameterEntryExceptionDecorators.entry_exception_pop_up_warning
     def add_k_v_row(self):
-        """ Add key, value row to parent row based on what row is highlighed in treeview"""
+        """ Add key, value row to parent row based on what row is highlighted in treeview"""
```
Comment (zlatko-minev) [inline qiskit_metal/_gui/widgets/create_component_window/parameter_entry_window.py:258]: Nice :)

File: qiskit_metal/_gui/widgets/create_component_window/parameter_entry_window.py [hunk 3]
```diff
@@ -292,8 +292,9 @@ def generate_model_data(self):
                 if param.default:
                     param_dict[param.name] = param.default
                 else:
+                    class_name = self.qcomp_class.__name__ if param.name == 'name' else None
                     param_dict[param.name] = create_default_from_type(
-                        param.annotation)
+                        param.annotation, param_name=class_name)
```
Comment (zlatko-minev) [inline qiskit_metal/_gui/widgets/create_component_window/parameter_entry_window.py:297]: Cool

File: qiskit_metal/_gui/widgets/create_component_window/parameter_entry_window.py (hunk 1 above)
Comment (zlatko-minev) [reply qiskit_metal/_gui/widgets/create_component_window/parameter_entry_window.py:449]: you mean this is for when we add a new row to a dictionary? Yes, sounds good

**Review by zlatko-minev — DISMISSED** [review]

File: qiskit_metal/_gui/widgets/create_component_window/parameter_entry_window.py [hunk 4]
```diff
@@ -442,8 +443,10 @@ class definition
                 return memtup[1]
 
 
-def create_default_from_type(my_t: type):
+def create_default_from_type(my_t: type, param_name=None):
     """Create default values for a given type"""
```
Comment (JeremyDrysdale-ibm) [inline qiskit_metal/_gui/widgets/create_component_window/parameter_entry_window.py:447]: Please add function args to the docstring.

**Review by JeremyDrysdale-ibm — CHANGES_REQUESTED** [review]

**Comment by grace-harper-ibm** [comment]
Regarding Zlatko's above comment, this `if param_name is not None:` is currently used only to rename the default "name" argument to be, for example, TransmonPocket-654 or NGon-342

**Review by JeremyDrysdale-ibm — APPROVED** [review]

**Review by marcolincs — APPROVED** [review]
thank you for the changes

**Pull request merged**